components/MultiPanel/index.ts

```diff
@@ -1,12 +1,183 @@
+
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newHeading = this._nextHeading();
+        break;
+
+      case 'Home':
+        newHeading = this._firstHeading();
+        break;
+
+      case 'End':
+        newHeading = this._lastHeading();
+        break;
+
+      case 'Enter':
+      case ' ':
+        if (this._isHeading(document.activeElement)) {
+            this._expand(document.activeElement);
+        }
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    currentHeading.setAttribute('tabindex', '-1');
+    if (newHeading) {
+      newHeading.setAttribute('tabindex', '0');
+      newHeading.focus();
+    }
+  }
+
+  _expand (heading : Element) {
+    const content = heading.nextElementSibling;
+    
+    // heading elment should always have nextElementSibling (checked on _childrenChange)
+    // but in case it is null, return.
+    if (content === null) {
+      return;
+    }
+
+    // toggle expanded and aria-expanded attoributes
+    if (content.hasAttribute('expanded')) {
+      content.removeAttribute('expanded')
+      content.setAttribute('aria-expanded', 'false');
+    } else {
+      content.setAttribute('expanded', '') 
+      content.setAttribute('aria-expanded', 'true');  
+    }
+  }
+  
+  // children of multi-panel should always in even nuber (heading/content pair)
+  // if children are odd numbers, add a div at the end to prevent potential error.
+  _childrenChange () {
+    if (this.children.length % 2 !== 0) {
+      console.error('detected odd number of elements inside multi-panel, please make sure you have heading/content pair')
+      this.appendChild(document.createElement('div'))
+    }
+  }
+
+  // called to check if an element passed is panel heading
+  _isHeading (heading : Element) {
+    return heading.classList.contains('panel-heading');
+  }
+
+  // return list of panel heading elements
+  _allHeadings () {
+    return Array.from(this.querySelectorAll('.panel-heading'));
+  }
+
+  // returns headding that is before currently selected one.
+  _prevHeading () {
+    const headings : Element[] = this._allHeadings();
+    let newIdx : number = headings.findIndex(headings => headings === document.activeElement) - 1;
+    return headings[(newIdx + headings.length) % headings.length];
+  }
+
+  // returns headding that is after currently selected one.
+  _nextHeading () {
+    const headings : Element[] = this._allHeadings();
+    let newIdx  : number = headings.findIndex(heading => heading === document.activeElement) + 1;
+    return headings[newIdx % headings.length];
+  }
+
+  // returns first heading in multi-panel.
+  _firstHeading () {
+    const headings : Element[] = this._allHeadings();
+    return headings[0];
+  } 
+
+  // returns last heading in multi-panel.
+  _lastHeading () {
+    const headings : Element[] = this._allHeadings();
+    return headings[headings.length - 1];
```

Comment:
I think this would be faster:

```ts
const children = this.children;
if (children.length % 1) {
  // Odd number of children. Last one must be a heading.
  return children[children.length - 1];
}
// Otherwise, the second-last child is the heading.
return children[children.length - 2];
```